draco/asp/helpers.lp

```diff
@@ -4,3 +4,36 @@
 attribute(mark_channel,M,C) :-
     entity(encoding,M,E),
     attribute((encoding,channel),E,C).
+
+% @helper(mark_scale_channel) Whether a mark has an scale for a channel.
+attribute(mark_scale_channel,M,S,C) :-
+    entity(mark,V,M),
+    entity(scale,V,S),
+    attribute((scale,channel),S,C).
+attribute(mark_scale_channel,M,S,C) :-
+    entity(view,R,V),
+    entity(mark,V,M),
+    entity(scale,R,S),
+    attribute((scale,channel),S,C).
+
+% @helper(extent) extent of a field.
+attribute(extent,F,MIN,MAX) :-
+    attribute((field,min),F,MIN),
+    attribute((field,max),F,MAX).
+
+% @helper(discrete) Whether the channel is discrete.
+attribute(discrete,M,C) :-
```

Comment:
replace with `attribute(channel_type,V,(continuous;discrete))`. Note that V is a view, because I think the type is the same for the view and not just a mark. We can add another helper for a mark but I wonder whether it's better to write the constraints over the view.

Reply:
I'm a bit confused here. A scale has multiple channels, right? And the scales can be written in different levels(root/view)

Reply:
A scale always belongs to one channel. A scale is defined on a view but if there is only one view, we can make it implicit and define everything on the root. At least that's how I imagined it until now but I'm happy to consider alternatives.

Reply:
I mean a view can have multiple scales, one for each channel. 

By "different levels", I mean in the dict format a scale can be at the same level of the mark, but if there are multiple views where the marks share a scale, then the scale can be written at the same level as the views. 

This helper was written for the `line_area_with_discrete`(x and y channels cannot both be discrete) constraint. 
I think this constraint indicates:
 1) line and area should have both x and y encoding channels 
 2) each channel should have a corresponding scale 
 3) x and y scales cannot be both discrete. 

Maybe I should break this into 3 constraints, and the second one can be shared by similar constraints like `bar_tick_continuous_x_y` that relating marks and scales?

Reply:
I think the separate constraints make sense. 

For discrete vs continuous, we need to be a bit careful. A channel may have a linear scale but still be discrete if e.g. the channel is binned. We could get around this issue by a) adding new scale types (e.g. bin-linear) or b) having an encoding type that is discrete/continuous and that needs to be consistent with the scale type and possible transformations.